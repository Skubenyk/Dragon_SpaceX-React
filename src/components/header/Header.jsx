//*Header

//*Import base
import styles from './header.module.scss';
import { Link } from 'react-router-dom';
import classNames from 'classnames';
import { useState } from 'react';

//*Import image
import logo from './../../assets/logo/SpaceX_Crew-1_logo.svg';

//*Import components
import Authorization from './../authorization/Authorization';
import Registration from './../registration/Registration';
import Burger from './../burger/Burger';

const Header = () => {
  //*Hook useState for active
  const [modalActive, setmodalActive] = useState(false);
  const [authorizationActive, setauthorizationActive] = useState(false);
  const [burgerShow, setburgerShow] = useState(false);
  return (
    <>
      {/* Header section */}
      <header className={styles.header}>
        <div className={styles.headerWrapper}>
          {/* Logo */}
          <div className={styles.headerLogo}>
            <Link className={styles.logo} to='/'>
              <img className={styles.logoImage} src={logo} alt={'logo'} />
            </Link>
          </div>
          <div className={styles.headerBlock}>
            <div
              //*Сlosing burger menu with a click anywhere
              onClick={() => setburgerShow(false)}
              //*Сonditions for switching styles in mode active
              className={
                burgerShow
                  ? classNames(styles.headerNavigation, styles.show)
                  : styles.headerNavigation
              }
            >
              {/* Navigation */}
              <nav className={styles.navigation} data-testid='navigation'>
                <ul className={styles.navigationList}>
                  <li className={styles.navigationItem}>
                    <Link className={styles.navigationLink} to='/'>
                      ГOЛOВНА
                    </Link>
                  </li>
                  <li className={styles.navigationItem}>
                    <Link
                      className={styles.navigationLink}
                      to='/favorites'
                      data-testid='favorites-link'
                    >
                      ОБРАНЕ
                    </Link>
                  </li>
                  <li className={styles.navigationItem}>
                    <Link
                      className={styles.navigationLink}
                      to='/profile'
                      data-testid='home-link'
                    >
                      ПРОФІЛЬ
                    </Link>
                  </li>
                </ul>
              </nav>
            </div>
            {/* Buttons */}
            <div className={styles.headerOther}>
              <button
                //*Open authorization form
                onClick={() => setauthorizationActive(true)}
                className={classNames(styles.button, styles.buttonTransparent)}
                type='button'
              >
                <span>ВХІД</span>
              </button>
              <button
                //*Open registration form
                onClick={() => setmodalActive(true)}
                className={styles.button}
                type='button'
              >
                <span>РЕЄСТРАЦІЯ</span>
              </button>
            </div>
            {/* Burger menu */}
            <Burger active={burgerShow} setActive={setburgerShow} />
          </div>
        </div>
      </header>
      {/* Authorization form */}
      <Authorization
        active={authorizationActive}
        setActive={setauthorizationActive}
      />
      {/* Registration form */}
      <Registration active={modalActive} setActive={setmodalActive} />
    </>
  );
};

export default Header;
