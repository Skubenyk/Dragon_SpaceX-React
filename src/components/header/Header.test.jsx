import { render, screen } from '@testing-library/react';
import { MemoryRouter as Router } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import Header from './Header';
import { createMemoryHistory } from 'history';

describe('Header component', () => {
  it('Header renders', () => {
    render(
      <Router>
        <Header />
      </Router>
    );
    screen.debug();
  });

  it('renders обране link', () => {
    render(
      <Router>
        <Header />
      </Router>
    );
    const linkElement = screen.getByText(/обране/i);
    expect(linkElement).toBeInTheDocument();
  });

  it('Header empty snapshot', () => {
    render(
      <Router>
        <Header />
      </Router>
    );
    const header = render(
      <Router>
        <Header />
      </Router>
    );
    expect(header).toMatchSnapshot();
  });

  it('Header renders component', async () => {
    render(
      <Router>
        <Header />
      </Router>
    );
    await screen.findByTestId('firstName');
    screen.debug();
    userEvent.type(screen.getByTestId('firstName'), 'Євген');
    expect(screen.getByTestId('firstName')).toBeInTheDocument();
    screen.debug();
  });
});

describe('React router', () => {
  it('shold navigate to favorites', () => {
    const history = createMemoryHistory();
    const { container, getByTestId } = render(
      <Router history={history}>
        <Header />
      </Router>
    );
    const navigation = getByTestId('navigation');
    const link = getByTestId('favorites-link');
    expect(container.innerHTML).toMatch('ОБРАНЕ');
    expect(navigation).toContainElement(link);
  });

  it('shold navigate to home', () => {
    const history = createMemoryHistory();
    const { container, getByTestId } = render(
      <Router history={history}>
        <Header />
      </Router>
    );
    const navigation = getByTestId('navigation');
    const link = getByTestId('home-link');
    expect(container.innerHTML).toMatch('ГOЛOВНА');
    expect(navigation).toContainElement(link);
  });
});
