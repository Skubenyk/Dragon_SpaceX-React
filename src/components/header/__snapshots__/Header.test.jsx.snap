// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Header empty snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <div
          class="headerWrapper"
        >
          <div
            class="headerLogo"
          >
            <a
              class="logo"
              href="/"
            >
              <img
                alt="logo"
                class="logoImage"
                src="SpaceX_Crew-1_logo.svg"
              />
            </a>
          </div>
          <div
            class="headerBlock"
          >
            <div
              class="headerNavigation"
            >
              <nav
                class="navigation"
                data-testid="navigation"
              >
                <ul
                  class="navigationList"
                >
                  <li
                    class="navigationItem"
                  >
                    <a
                      class="navigationLink"
                      href="/"
                    >
                      ГOЛOВНА
                    </a>
                  </li>
                  <li
                    class="navigationItem"
                  >
                    <a
                      class="navigationLink"
                      data-testid="favorites-link"
                      href="/favorites"
                    >
                      ОБРАНЕ
                    </a>
                  </li>
                  <li
                    class="navigationItem"
                  >
                    <a
                      class="navigationLink"
                      data-testid="home-link"
                      href="/profile"
                    >
                      ПРОФІЛЬ
                    </a>
                  </li>
                </ul>
              </nav>
            </div>
            <div
              class="headerOther"
            >
              <button
                class="button buttonTransparent"
                type="button"
              >
                <span>
                  ВХІД
                </span>
              </button>
              <button
                class="button"
                type="button"
              >
                <span>
                  РЕЄСТРАЦІЯ
                </span>
              </button>
            </div>
            <div
              class="headerBurger"
            >
              <button
                class="burger"
                type="button"
              >
                <span />
                <span />
                <span />
              </button>
            </div>
          </div>
        </div>
      </header>
      <div
        class="modal"
      >
        <form
          class="form"
          method="post"
        >
          <h2
            class="title"
          >
            АВТОРИЗАЦІЯ
          </h2>
          <div
            class="wrapper"
          >
            <label
              class="label"
              for="email"
            />
            <input
              class="input email"
              name="email"
              placeholder="Пошта"
              type="email"
              value=""
            />
            <label
              class="label"
              for="password"
            />
            <input
              class="input password"
              name="password"
              placeholder="Пароль"
              type="password"
              value=""
            />
            <div
              class="button"
            >
              <button
                class="buttonSubmit"
                type="submit"
              >
                <span>
                  ВХІД
                </span>
              </button>
            </div>
          </div>
        </form>
      </div>
      <div
        class="modal"
      >
        <form
          class="form"
          method="post"
        >
          <h2
            class="title"
          >
            РЕЄСТРАЦІЯ
          </h2>
          <div
            class="wrapper"
          >
            <label
              class="label"
              for="firstName"
            />
            <input
              class="input firstName"
              data-testid="firstName"
              name="firstName"
              placeholder="Ім'я"
              type="text"
              value=""
            />
            <label
              class="label"
              for="lastName"
            />
            <input
              class="input lastName"
              name="lastName"
              placeholder="Прізвище"
              type="text"
              value=""
            />
            <label
              class="label"
              for="userName"
            />
            <input
              class="input userName"
              name="userName"
              placeholder="Ім'я користувача"
              type="text"
              value=""
            />
            <label
              class="label"
              for="email"
            />
            <input
              class="input email"
              name="email"
              placeholder="Пошта"
              type="email"
              value=""
            />
            <label
              class="label"
              for="password"
            />
            <input
              class="input password"
              name="password"
              placeholder="Пароль"
              type="password"
              value=""
            />
            <div
              class="button"
            >
              <button
                class="buttonSubmit"
                type="submit"
              >
                <span>
                  РЕЄСТРАЦІЯ
                </span>
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
    <div>
      <header
        class="header"
      >
        <div
          class="headerWrapper"
        >
          <div
            class="headerLogo"
          >
            <a
              class="logo"
              href="/"
            >
              <img
                alt="logo"
                class="logoImage"
                src="SpaceX_Crew-1_logo.svg"
              />
            </a>
          </div>
          <div
            class="headerBlock"
          >
            <div
              class="headerNavigation"
            >
              <nav
                class="navigation"
                data-testid="navigation"
              >
                <ul
                  class="navigationList"
                >
                  <li
                    class="navigationItem"
                  >
                    <a
                      class="navigationLink"
                      href="/"
                    >
                      ГOЛOВНА
                    </a>
                  </li>
                  <li
                    class="navigationItem"
                  >
                    <a
                      class="navigationLink"
                      data-testid="favorites-link"
                      href="/favorites"
                    >
                      ОБРАНЕ
                    </a>
                  </li>
                  <li
                    class="navigationItem"
                  >
                    <a
                      class="navigationLink"
                      data-testid="home-link"
                      href="/profile"
                    >
                      ПРОФІЛЬ
                    </a>
                  </li>
                </ul>
              </nav>
            </div>
            <div
              class="headerOther"
            >
              <button
                class="button buttonTransparent"
                type="button"
              >
                <span>
                  ВХІД
                </span>
              </button>
              <button
                class="button"
                type="button"
              >
                <span>
                  РЕЄСТРАЦІЯ
                </span>
              </button>
            </div>
            <div
              class="headerBurger"
            >
              <button
                class="burger"
                type="button"
              >
                <span />
                <span />
                <span />
              </button>
            </div>
          </div>
        </div>
      </header>
      <div
        class="modal"
      >
        <form
          class="form"
          method="post"
        >
          <h2
            class="title"
          >
            АВТОРИЗАЦІЯ
          </h2>
          <div
            class="wrapper"
          >
            <label
              class="label"
              for="email"
            />
            <input
              class="input email"
              name="email"
              placeholder="Пошта"
              type="email"
              value=""
            />
            <label
              class="label"
              for="password"
            />
            <input
              class="input password"
              name="password"
              placeholder="Пароль"
              type="password"
              value=""
            />
            <div
              class="button"
            >
              <button
                class="buttonSubmit"
                type="submit"
              >
                <span>
                  ВХІД
                </span>
              </button>
            </div>
          </div>
        </form>
      </div>
      <div
        class="modal"
      >
        <form
          class="form"
          method="post"
        >
          <h2
            class="title"
          >
            РЕЄСТРАЦІЯ
          </h2>
          <div
            class="wrapper"
          >
            <label
              class="label"
              for="firstName"
            />
            <input
              class="input firstName"
              data-testid="firstName"
              name="firstName"
              placeholder="Ім'я"
              type="text"
              value=""
            />
            <label
              class="label"
              for="lastName"
            />
            <input
              class="input lastName"
              name="lastName"
              placeholder="Прізвище"
              type="text"
              value=""
            />
            <label
              class="label"
              for="userName"
            />
            <input
              class="input userName"
              name="userName"
              placeholder="Ім'я користувача"
              type="text"
              value=""
            />
            <label
              class="label"
              for="email"
            />
            <input
              class="input email"
              name="email"
              placeholder="Пошта"
              type="email"
              value=""
            />
            <label
              class="label"
              for="password"
            />
            <input
              class="input password"
              name="password"
              placeholder="Пароль"
              type="password"
              value=""
            />
            <div
              class="button"
            >
              <button
                class="buttonSubmit"
                type="submit"
              >
                <span>
                  РЕЄСТРАЦІЯ
                </span>
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <div
        class="headerWrapper"
      >
        <div
          class="headerLogo"
        >
          <a
            class="logo"
            href="/"
          >
            <img
              alt="logo"
              class="logoImage"
              src="SpaceX_Crew-1_logo.svg"
            />
          </a>
        </div>
        <div
          class="headerBlock"
        >
          <div
            class="headerNavigation"
          >
            <nav
              class="navigation"
              data-testid="navigation"
            >
              <ul
                class="navigationList"
              >
                <li
                  class="navigationItem"
                >
                  <a
                    class="navigationLink"
                    href="/"
                  >
                    ГOЛOВНА
                  </a>
                </li>
                <li
                  class="navigationItem"
                >
                  <a
                    class="navigationLink"
                    data-testid="favorites-link"
                    href="/favorites"
                  >
                    ОБРАНЕ
                  </a>
                </li>
                <li
                  class="navigationItem"
                >
                  <a
                    class="navigationLink"
                    data-testid="home-link"
                    href="/profile"
                  >
                    ПРОФІЛЬ
                  </a>
                </li>
              </ul>
            </nav>
          </div>
          <div
            class="headerOther"
          >
            <button
              class="button buttonTransparent"
              type="button"
            >
              <span>
                ВХІД
              </span>
            </button>
            <button
              class="button"
              type="button"
            >
              <span>
                РЕЄСТРАЦІЯ
              </span>
            </button>
          </div>
          <div
            class="headerBurger"
          >
            <button
              class="burger"
              type="button"
            >
              <span />
              <span />
              <span />
            </button>
          </div>
        </div>
      </div>
    </header>
    <div
      class="modal"
    >
      <form
        class="form"
        method="post"
      >
        <h2
          class="title"
        >
          АВТОРИЗАЦІЯ
        </h2>
        <div
          class="wrapper"
        >
          <label
            class="label"
            for="email"
          />
          <input
            class="input email"
            name="email"
            placeholder="Пошта"
            type="email"
            value=""
          />
          <label
            class="label"
            for="password"
          />
          <input
            class="input password"
            name="password"
            placeholder="Пароль"
            type="password"
            value=""
          />
          <div
            class="button"
          >
            <button
              class="buttonSubmit"
              type="submit"
            >
              <span>
                ВХІД
              </span>
            </button>
          </div>
        </div>
      </form>
    </div>
    <div
      class="modal"
    >
      <form
        class="form"
        method="post"
      >
        <h2
          class="title"
        >
          РЕЄСТРАЦІЯ
        </h2>
        <div
          class="wrapper"
        >
          <label
            class="label"
            for="firstName"
          />
          <input
            class="input firstName"
            data-testid="firstName"
            name="firstName"
            placeholder="Ім'я"
            type="text"
            value=""
          />
          <label
            class="label"
            for="lastName"
          />
          <input
            class="input lastName"
            name="lastName"
            placeholder="Прізвище"
            type="text"
            value=""
          />
          <label
            class="label"
            for="userName"
          />
          <input
            class="input userName"
            name="userName"
            placeholder="Ім'я користувача"
            type="text"
            value=""
          />
          <label
            class="label"
            for="email"
          />
          <input
            class="input email"
            name="email"
            placeholder="Пошта"
            type="email"
            value=""
          />
          <label
            class="label"
            for="password"
          />
          <input
            class="input password"
            name="password"
            placeholder="Пароль"
            type="password"
            value=""
          />
          <div
            class="button"
          >
            <button
              class="buttonSubmit"
              type="submit"
            >
              <span>
                РЕЄСТРАЦІЯ
              </span>
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
